src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/StringSearchValues.cs

```diff
@@ -297,6 +312,68 @@ static SearchValues<string> PickAhoCorasickImplementation<TCaseSensitivity>(AhoC
             }
         }
 
+        private static bool TryGenerateAllCasePermutationsForPrefixes(ReadOnlySpan<string> values, int n, int maxValues, [NotNullWhen(true)] out string[]? newValues)
+        {
+            Debug.Assert(n is 2 or 3);
+            Debug.Assert(values.Length < maxValues);
+
+            // Count how many possible permutations there are.
+            int newValuesCount = 0;
+
+            foreach (string value in values)
+            {
+                int permutations = 1;
+
+                foreach (char c in value.AsSpan(0, n))
+                {
+                    Debug.Assert(char.IsAscii(c));
```

Comment:
What guarantees we're ascii only here?

Reply:
We check that earlier in `TryGetTeddyAcceleratedValues` before calling into `TryGenerateAllCasePermutationsForPrefixes`.
https://github.com/dotnet/runtime/blob/54e8e3d1f45ae17504638f661bd9369569299586/src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/StringSearchValues.cs#L215-L225